File: nan21.dnet.module.md/nan21.dnet.module.md.ui.extjs/src/main/resources/webapp/js/ds/DeliveryMethodLovDs.js
/* 
 * DNet eBusiness Suite
 * Copyright: Nan21 Electronics SRL. All rights reserved.
 * Use is subject to license terms.
 */

Ext.define("net.nan21.dnet.module.md.base.tx.ds.model.DeliveryMethodLovDs", {
	extend: 'Ext.data.Model',
	fields: [
		{name:"active", type:"boolean"},
		{name:"clientId", type:"int", useNull:true},
		{name:"delivery", type:"boolean"},
		{name:"description", type:"string"},
		{name:"externalCourier", type:"boolean"},
		{name:"id", type:"int", useNull:true},
		{name:"name", type:"string"}
	]
});

File: nan21.dnet.module.md/nan21.dnet.module.md.presenter/src/main/java/net/nan21/dnet/module/md/base/tx/ds/model/DeliveryMethodLovDs.java
/* 
 * DNet eBusiness Suite
 * Copyright: 2010-2013 Nan21 Electronics SRL. All rights reserved.
 * Use is subject to license terms.
 */
package net.nan21.dnet.module.md.base.tx.ds.model;

import net.nan21.dnet.core.api.annotation.Ds;
import net.nan21.dnet.core.api.annotation.DsField;
import net.nan21.dnet.core.api.annotation.SortField;
import net.nan21.dnet.core.presenter.model.base.AbstractTypeLov;
import net.nan21.dnet.module.md.base.tx.domain.entity.DeliveryMethod;

@Ds(entity = DeliveryMethod.class, jpqlWhere = " e.active = true ", sort = {@SortField(field = DeliveryMethodLovDs.f_name)})
public class DeliveryMethodLovDs extends AbstractTypeLov<DeliveryMethod> {

	public static final String f_delivery = "delivery";
	public static final String f_externalCourier = "externalCourier";

	@DsField
	private Boolean delivery;

	@DsField
	private Boolean externalCourier;

	public DeliveryMethodLovDs() {
		super();
	}

	public DeliveryMethodLovDs(DeliveryMethod e) {
		super(e);
	}

	public Boolean getDelivery() {
		return this.delivery;
	}

	public void setDelivery(Boolean delivery) {
		this.delivery = delivery;
	}

	public Boolean getExternalCourier() {
		return this.externalCourier;
	}

	public void setExternalCourier(Boolean externalCourier) {
		this.externalCourier = externalCourier;
	}
}
